torchvision/csrc/io/decoder/decoder.cpp

```diff
@@ -397,10 +395,10 @@ bool Decoder::init(
 }
 
 bool Decoder::openStreams(std::vector<DecoderMetadata>* metadata) {
-  for (int i = 0; i < inputCtx_->nb_streams; i++) {
+  for (int i = 0; i < static_cast<int> (inputCtx_->nb_streams); i++) {
```

Comment:
Could this cast from unsigned int to int lead to an issue? If the value exceeds (2^32 - 1) in unsigned int, casting it to signed int would make it negative, causing unexpected loop behaviour. I don't know the range of values `nb_streams` can have though. A safer way to do this would be to declare `i` as `unsigned int`.

Reply:
So so nb_streams, will always be a positive int, but stream_id (which we compare it to a few lines later) in principle won't. We can potentially remove the cast and leave the warning if that's preferable?

Reply:
That's fine, thanks!